src/EditorFeatures/Core/Implementation/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -253,12 +253,17 @@ private void RemoveDocumentFromWorkspace_NoLock(MetadataAsSourceGeneratedFileInf
             }
         }
 
-        private void InitializeWorkspace(Project project)
+        private void InitializeWorkspace(Project project, OptionSet option)
```

Comment:
This looked more cleaner but your question makes sense. Let me see if how that turns out